erpnext/public/js/controllers/taxes_and_totals.js

```diff
@@ -634,23 +639,46 @@ erpnext.taxes_and_totals = erpnext.payments.extend({
 		}
 	},
 
-	set_default_payment: function(total_amount_to_pay, update_paid_amount){
+	update_paid_amount_for_return: function() {
+		var me = this;
+		var grand_total = this.frm.doc.rounded_total || this.frm.doc.grand_total;
+
+		if(this.frm.doc.party_account_currency == this.frm.doc.currency) {
+			var total_amount_to_pay = flt((grand_total - this.frm.doc.total_advance
+				- this.frm.doc.write_off_amount), precision("grand_total"));
+		} else {
+			var total_amount_to_pay = flt(
+				(flt(grand_total*this.frm.doc.conversion_rate, precision("grand_total"))
+					- this.frm.doc.total_advance - this.frm.doc.base_write_off_amount),
+				precision("base_grand_total")
+			);
+		}
+
+		$.each(this.frm.doc['payments'] || [], function(index, data) {
```

Comment:
There can be multiple payment rows based on mode of payments. We don't know by which mode-of-payment they are returning the payment or even returning the amount or not. I think we should add a field in "POS Settings" for "Default mode-of-payment for return" and allocate the return amount against it.
Otherwise, we can also blank the entire payment table and remove "is_pos" check. In that case, we are assuming that we are issuing Credit note to customer, not refunding any cash.